agent/testagent.go

```diff
@@ -182,6 +182,12 @@ func (a *TestAgent) Start(t *testing.T) error {
 
 	portsConfig := randomPortsSource(t, a.UseTLS)
 
+	//a.Overrides = `
```

Comment:
We could turn this on for all agents but I'd recommend turning it off by default and just enabling it for the tests that need it.

Reply:
I think not having this is the cause of all the existing CI failures. However if you uncomment this then your new tests regarding checking the enforcement behavior of not allowing the PeerName field to be set would then fail.

I generally agree that the best course of action would be to update tests that do the nasty things to opt into the behavior.

Reply:
I think as of now, all tests that need this config option on have it so I should remove this commented block